src/scib_metrics/_lisi.py

```diff
@@ -0,0 +1,57 @@
+from typing import Tuple
+
+import numpy as np
+from scipy.sparse import csr_matrix
+from sklearn.utils import check_array
+
+from scib_metrics.utils import compute_simpson_index
+
+
+def _convert_knn_graph_to_idx(knn_graph: csr_matrix) -> Tuple[np.ndarray, np.ndarray]:
+    check_array(knn_graph, accept_sparse="csr")
+
+    n_neighbors = np.unique(knn_graph.nonzero()[0], return_counts=True)[1]
+    if len(np.unique(n_neighbors)) > 1:
+        raise ValueError("Each cell must have the same number of neighbors.")
+
+    n_neighbors = int(np.unique(n_neighbors)[0])
+    n_cells = knn_graph.shape[0]
+
+    nn_idx = np.empty(shape=(n_cells, n_neighbors), dtype=np.int32)
+    nn_dists = np.empty(shape=(n_cells, n_neighbors), dtype=np.float32)
+    nn_dists[:] = np.NaN
+    for i in range(n_cells):
+        knn_graph_row = knn_graph.getrow(i)
+        sorted_dist_idxs = np.argsort(knn_graph_row.data)
+        nn_idx[i, :] = knn_graph_row.indices[sorted_dist_idxs]
+        nn_dists[i, :] = knn_graph_row.data[sorted_dist_idxs]
```

Comment:
this can probably be sped up?

Reply:
https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NearestNeighbors.html#sklearn.neighbors.NearestNeighbors

can we use this with precomputed and then call kneighbors?